Bugzilla/WebService/User.pm

```diff
@@ -504,6 +508,67 @@ sub whoami {
   );
 }
 
+#
+# Saved Searches
+#
+
+sub add_saved_search {
+  state $check = compile(Object, Dict[name => Str, url => Str]);
+  my ($self, $params) = $check->(@_);
+
+  # The `create()` method fails when a saved search with the same name exists.
+  # The UI should ask in advance if the user wants to override the existing one,
+  # and if the answer is yes, use `update_saved_search()` instead.
+  my $search = Bugzilla::Search::Saved->create({
+    name => $params->{name}, query => $params->{url}, link_in_footer => 1
+  });
+
+  return $self->_query_to_hash($search);
```

Comment:
Since we are creating new API methods should always return a hash with a keyword containing the results. We should try to standardize on all result structures where possible.

For example in this case it would be:

```json
{
  "queries": [
      {
         "id": 1,
         "name": "foo",
         "url": "somelongurl",
      }
   ]
}
```

Reply:
I think newer methods like this don’t have to follow the legacy, redundant structure, because it’s only available via the REST API and only for internal use (undocumented). The [Component Watching API](https://github.com/mozilla-bteam/bmo/blob/master/extensions/ComponentWatching/lib/WebService.pm) I’ve added recently is also not following the tradition. We are moving to GraphQL sooner or later, so the structure here doesn’t matter 🙂

Reply:
> 
> 
> I think newer methods like this don’t have to follow the legacy, redundant structure, because it’s only available via the REST API and only for internal use (undocumented) for now. The [Component Watching API](https://github.com/mozilla-bteam/bmo/blob/master/extensions/ComponentWatching/lib/WebService.pm) I’ve added recently is also not following the tradition. We are moving to GraphQL sooner or later, so the structure here doesn’t matter 🙂

Not a showstopper for this, but I politely disagree on two points. 1) We should write all APIs as if it could also be used for the general public and not just for internal only reasons. Bugzilla has always lacked in a robust API that could be used to do most actions. Anything we can do to make this better at the same time as fulfilling our own needs the better. And 2) working with the non-obsolete API methods in Phabricator, it is nice to always expect a consistent return data structure and not have to guess at what to expect to be returned. They have been working to deprecate the old API methods that do not adhere to the consistent structure. Also they had to change it to allow for the paging functionality built in. We do not want to put out new methods that people start to use and then have to change them later. Better to do it right now.

I am not aware or knowledgeable about GraphQL yet so you are probably right that the current code may not matter. But even so it would be nice if when we do move over the clients do not have to change very much about how they receive the new data in there code. Would be cool if it all just looked the same, just coming from something different on the backend.